assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -355,21 +460,22 @@ const baseSelectors = {
 		if ( registryKey === undefined ) {
 			return undefined;
 		}
+
 		// Sorting the modules object by order property.
 		const sortedModules = sortByProperty( Object.values( modules ), 'order' );
 		const mappedModules = sortedModules.map( ( module ) => {
 			const moduleWithComponent = { ...module };
-			if ( ModuleComponents[ registryKey ] ) {
-				// If there is a settingsComponent that was passed use it, otherwise set to the default.
-				if ( ModuleComponents[ registryKey ][ module.slug ] ) {
-					moduleWithComponent.settingsComponent = ModuleComponents[ registryKey ][ module.slug ];
-				} else {
-					moduleWithComponent.settingsComponent = DefaultModuleSettings;
-				}
+
+			// If there is a settingsComponent that was passed use it, otherwise set to the default.
+			if ( ModuleComponents[ registryKey ] && ModuleComponents[ registryKey ][ module.slug ] ) {
+				moduleWithComponent.settingsComponent = ModuleComponents[ registryKey ][ module.slug ];
+			} else {
+				moduleWithComponent.settingsComponent = undefined;
 			}
```

Comment:
This shouldn't be changed from what it was before. This is related to my other comment: `DefaultModuleSettings` should have exactly the purpose indicated by its name - it should be the default module settings component. The component implemented further above should rather be `SimpleModuleSettings` or something like that.